cpp/src/arrow/python/pyarrow.h

```diff
@@ -45,66 +45,34 @@ namespace py {
-
-ARROW_PYTHON_EXPORT bool is_array(PyObject* array);
-ARROW_PYTHON_EXPORT Status unwrap_array(PyObject* array, std::shared_ptr<Array>* out);
-ARROW_PYTHON_EXPORT PyObject* wrap_array(const std::shared_ptr<Array>& array);
-ARROW_PYTHON_EXPORT PyObject* wrap_chunked_array(
-    const std::shared_ptr<ChunkedArray>& array);
-
-ARROW_PYTHON_EXPORT bool is_tensor(PyObject* tensor);
-ARROW_PYTHON_EXPORT Status unwrap_tensor(PyObject* tensor, std::shared_ptr<Tensor>* out);
-ARROW_PYTHON_EXPORT PyObject* wrap_tensor(const std::shared_ptr<Tensor>& tensor);
-
-ARROW_PYTHON_EXPORT bool is_sparse_coo_tensor(PyObject* sparse_tensor);
-ARROW_PYTHON_EXPORT Status
-unwrap_sparse_coo_tensor(PyObject* sparse_tensor, std::shared_ptr<SparseCOOTensor>* out);
-ARROW_PYTHON_EXPORT PyObject* wrap_sparse_coo_tensor(
-    const std::shared_ptr<SparseCOOTensor>& sparse_tensor);
-
-ARROW_PYTHON_EXPORT bool is_sparse_csr_matrix(PyObject* sparse_tensor);
-ARROW_PYTHON_EXPORT Status
-unwrap_sparse_csr_matrix(PyObject* sparse_tensor, std::shared_ptr<SparseCSRMatrix>* out);
-ARROW_PYTHON_EXPORT PyObject* wrap_sparse_csr_matrix(
-    const std::shared_ptr<SparseCSRMatrix>& sparse_tensor);
-
-ARROW_PYTHON_EXPORT bool is_sparse_csc_matrix(PyObject* sparse_tensor);
-ARROW_PYTHON_EXPORT Status
-unwrap_sparse_csc_matrix(PyObject* sparse_tensor, std::shared_ptr<SparseCSCMatrix>* out);
-ARROW_PYTHON_EXPORT PyObject* wrap_sparse_csc_matrix(
-    const std::shared_ptr<SparseCSCMatrix>& sparse_tensor);
-
-ARROW_PYTHON_EXPORT bool is_table(PyObject* table);
-ARROW_PYTHON_EXPORT Status unwrap_table(PyObject* table, std::shared_ptr<Table>* out);
-ARROW_PYTHON_EXPORT PyObject* wrap_table(const std::shared_ptr<Table>& table);
-
-ARROW_PYTHON_EXPORT bool is_sparse_csf_tensor(PyObject* sparse_tensor);
-ARROW_PYTHON_EXPORT Status
-unwrap_sparse_csf_tensor(PyObject* sparse_tensor, std::shared_ptr<SparseCSFTensor>* out);
-ARROW_PYTHON_EXPORT PyObject* wrap_sparse_csf_tensor(
-    const std::shared_ptr<SparseCSFTensor>& sparse_tensor);
-
-ARROW_PYTHON_EXPORT bool is_record_batch(PyObject* batch);
-ARROW_PYTHON_EXPORT Status unwrap_record_batch(PyObject* batch,
-                                               std::shared_ptr<RecordBatch>* out);
-ARROW_PYTHON_EXPORT PyObject* wrap_record_batch(
-    const std::shared_ptr<RecordBatch>& batch);
+#define DECLARE_WRAP_FUNCTIONS(FUNC_SUFFIX, TYPE_NAME)                                 \
+  ARROW_PYTHON_EXPORT bool is_##FUNC_SUFFIX(PyObject*);                                \
+  ARROW_PYTHON_EXPORT Result<std::shared_ptr<TYPE_NAME>> unwrap_##FUNC_SUFFIX(         \
+      PyObject*);                                                                      \
+  ARROW_PYTHON_EXPORT PyObject* wrap_##FUNC_SUFFIX(const std::shared_ptr<TYPE_NAME>&); \
+  ARROW_DEPRECATED("Use Result-returning version")                                     \
+  ARROW_PYTHON_EXPORT Status unwrap_##FUNC_SUFFIX(PyObject*,                           \
```

Comment:
Is it necessary to deprecate these instead of just removing them? Aren't they internal?

Reply:
I don't know. @wesm or @xhochy could say.

Reply:
They are public APIs https://github.com/blue-yonder/turbodbc/blob/32388891802f77a2610e38204060e946438221d4/cpp/turbodbc_arrow/Library/src/set_arrow_parameters.cpp#L427

Reply:
Ok. In this case, I've renamed the `XXX_record_batch` APIs to `XXX_batch`, for consistency with the Cython API. Is that a problem?